string mode
float64 desired_distance
---
bool success
